blosc/blosc2.h

```diff
@@ -711,6 +734,31 @@ BLOSC_EXPORT void blosc_cbuffer_versions(const void* cbuffer, int* version,
  */
 BLOSC_EXPORT const char* blosc_cbuffer_complib(const void* cbuffer);
 
+/*********************************************************************
+  Structures and functions related with user-defined input/output.
+*********************************************************************/
+
+typedef void*  (*blosc2_open_cb)(const char *urlpath, const char *mode, void *params);
+typedef int    (*blosc2_close_cb)(void *stream, void* params);
+typedef int    (*blosc2_seek_cb)(void *stream, int64_t, int, void* params);
+typedef size_t (*blosc2_write_cb)(const void *ptr, size_t size, size_t nitems, void *stream, void *params);
+typedef size_t (*blosc2_read_cb)(void *ptr, size_t size, size_t nitems, void *stream, void *params);
+
+typedef struct {
+  blosc2_open_cb open;
+  blosc2_close_cb close;
+  blosc2_seek_cb seek;
+  blosc2_write_cb write;
+  blosc2_read_cb read;
+} blosc2_io;
+
+static const blosc2_io BLOSC2_IO_DEFAULTS = {
+  .open = (blosc2_open_cb) fopen,
```

Comment:
Because of the integer type problem in `fseek` for `offset` it would be better to have a custom wrapper of fseek that is called. For example, on 32-bit Windows when `fseek` is called it will still use `long int` which I believe is 4 bytes and so the function signature will not match with `blosc2_seek_cb`.

It might be best to do this for all the FILE functions. Even if they only wrap a single call. That way you can separate your interface from the interface of the FILE functions which vary on the different platforms. 

https://github.com/zlib-ng/minizip-ng/blob/master/mz_strm_os_posix.c

Reply:
I have implemented it in f5846e5. Is it what you suggested?